src/algorithms/motion_tracking/MotionTrackingWrapper.py

```diff
@@ -0,0 +1,211 @@
+        self.is_first_day = True
+
+        self.morning_intensity = (
+            50.0  # Fixed at a dim level at which CV still functions well.
+        )
+        self.target_intensity = 110.0  # Its optimal value depends on plant species, developmental stage, and environmental factors. Tuned by the RL agent daily.
+        self.sensitivity = uema(
+            alpha=0.1
+        )  # = (change in intensity) / (change in plants openness). Adjusted daily by motion tracker to attempt to reach target_intensity at max openness.
+
+        self.mean_areas = defaultdict(float)
+        self.openness_record = []
+        self.openness_trace = uema(alpha=0.1)
+
+        # RL agent stuff
+        self.agent_started = False
+        self.agent_Imin = 50.0
+        self.agent_Imax = 150.0
+        self.agent_increment = 10.0
+
+    def get_action(self) -> float:
+        if self.is_first_day:
+            return self.target_intensity
+        else:
+            openness = self.openness_trace.compute().item()
+            proposed_action = (
+                self.morning_intensity
+                + self.sensitivity.compute().item() * np.log(100 * openness + 1)
+            )
+            return max(min(proposed_action, self.agent_Imax), self.agent_Imin)
+
+    def adjust_sensitivity(self):
+        max_openness = np.mean(np.sort(self.openness_record)[-5:])
+        proposed_sensitivity = (
+            self.target_intensity - self.morning_intensity
+        ) / np.log(100 * max_openness + 1)
+        self.sensitivity.update(proposed_sensitivity)
+        logger.info(
+            f"Set target_intensity = {self.target_intensity}. Set sensitivity = {self.sensitivity.compute().item():.2f}."
+        )
+
+    def reward(self):
+        if self.env_local_time is not None:
+            current_time = self.env_local_time.replace(second=0, microsecond=0)
+            current_area = self.mean_areas.get(current_time, -1)
+            yesterday_area = self.mean_areas.get(current_time - timedelta(days=1), -1)
+            return current_area / yesterday_area - 1
+        else:
+            return 0.0
+
+    async def start(self, observation: np.ndarray, extra: Dict[str, Any]):  # type: ignore
+        self.openness_trace.reset()
+        self.openness_record = []
+
+        self.env_local_time = observation[0]
+        mean_area = observation[1]
+        self.mean_areas[self.env_local_time.replace(second=0, microsecond=0)] = float(
+            mean_area
+        )
+
+        if self.is_night():
+            action = 0.0
+        elif self.is_zeroth_tod():
+            action = self.morning_intensity
+        else:
+            logger.warning(
+                f"Starting motion-tracking controller during daytime. Enforce target_intensity = {self.target_intensity} until tomorrow."
+            )
+            action = self.target_intensity
+
+        return action, {}
+
+    async def step(self, reward: float, observation: np.ndarray, extra: Dict[str, Any]):
```

Comment:
this function is 76 lines long, consider breaking it into smaller functions for readability